src/main/webapp/definitions/alfresco-content-models.yaml

```diff
@@ -0,0 +1,738 @@
+          An integer describing the total number of entities in the collection.
+          The API might not be able to determine this value,
+          in which case this property will not be present.
+      skipCount:
+        type: integer
+        format: int64
+        description: |
+          An integer describing how many entities exist in the collection before
+          those included in this list. If there was no **skipCount** parameter then the 
+          default value is 0.
+      maxItems:
+        type: integer
+        format: int64
+        description: |
+          The value of the **maxItems** parameter used to generate this list.
+          If there was no **maxItems** parameter then the default value is 100.
+  QNamePaging:
+    type: object
+    properties:
+      list:
+        type: object
+        properties:
+          pagination:
+            $ref: '#/definitions/Pagination'
+          entries:
+            type: array
+            items:
+              $ref: '#/definitions/QNameEntry'
+          source:
+            $ref: '#/definitions/QName'
+  QNameEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/QName'
+  QName:
+    type: object
+    required:
+      - namespacePrefix
+      - namespaceUri
+      - name
+    properties:
+      namespacePrefix:
+        type: string
+      namespaceUri:
+        type: string
+      name:
+        type: string
+  ModelEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Model'
+  Model:
+    allOf:
+      - $ref: '#/definitions/QName'
+      - properties:
+          version:
+            description: the model version
+            type: string
+          author:
+            description: author of the model
+            type: string
+          description:
+            description: the model description
+            type: string
+  TypeEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Type'
+  Type:
+    allOf:
+      - $ref: '#/definitions/QName'
+      - properties:
+          title:
+            description: the human-readable type title
+            type: string
+          parent:
+            description: parent type defination
+            $ref: '#/definitions/Type'
+          description:
+            description: description of the type
+            type: string
+          archive:
+            description: define if the type is archived on delete
+            type: boolean
+  AspectEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Aspect'
+  Aspect:
+    allOf:
+      - $ref: '#/definitions/QName'
+  ConstraintEntry:
+    type: object
+    required:
+      - entry
+    properties:
+      entry:
+        $ref: '#/definitions/Constraint'
+  Constraint:
+    allOf:
+      - $ref: '#/definitions/QName'
```

Comment:
Constraint isn't a QName, see https://github.com/Alfresco/alfresco-data-model/blob/c7aaa53cce8245889ef79daffac46b53f827ee0e/src/main/java/org/alfresco/service/cmr/dictionary/Constraint.java

Reply:
I and Giovanni are associating a QName to the Constraint because we are referring to the ConstraintDefinition class that has a QName. We should call the entity ConstraintDefinition instead of Constraint?

https://github.com/Alfresco/alfresco-data-model/blob/27a75808f14de41154b7a817599b004d88d86e8d/src/main/java/org/alfresco/service/cmr/dictionary/ConstraintDefinition.java#L48

Reply:
Ah yes, I see what you mean, however a constraint has more than just a QName, it also has a title and description like Type. There's also a getConstraint() method on there which has some additional data not reflected here. 

We don't have to (and shouldn't) match the underlying API exactly so we can collapse everything into one definition, I think "Constraint" as a name is fine.

It's also much easier once we have a prototype working to see it for real, sometimes it's hard to visualise at this level.

Reply:
As I can remember from the CMM work, an application requires all the read properties of the Constraint class, i.e. type, title, description, parameters. The `shortName` is actually the `prefixedQName` of the constraint.